docs/src/main/asciidoc/_high-availability/cloud/Kubernetes.adoc

```diff
@@ -0,0 +1,236 @@
+Thus, existing applications need to update the default cache mode to optimize scalability.
+
+Applications built using Maven with https://docs.wildfly.org/wildfly-maven-plugin/releases/4.2/package-mojo.html#packagingScripts[WildFly Maven Plugin]
+to create an image can be easily configured to update the default cache mode.
+This can be done by configuring a set commands in the packaging step to reconfigure the bundled server.
+The following `infinispan.cli` configuration script will reconfigure the `infinispan` subsystem to use the distributed cache for web sessions:
+
+// n.b. regarding the recommended configuration:
+// we are intentionally leave out the file store (spin a new instance when needed) + leave out locking and transactions.
+// The provided expiration configuration saves unnecessary 1 expiration thread that would otherwise run.
+
+[source,cli,title=infinispan.cli]
+----
+/subsystem=infinispan/cache-container=web/distributed-cache=sessions:add
+/subsystem=infinispan/cache-container=web/distributed-cache=sessions/component=expiration:add(interval=0)
+/subsystem=infinispan/cache-container=web:write-attribute(name=default-cache,value=sessions)
+----
+
+The Maven plugin configuration can then be updated to update the generated container image:
+
+[source,xml]
+----
+<plugin>
+    <groupId>org.wildfly.plugins</groupId>
+    <artifactId>wildfly-maven-plugin</artifactId>
+    <executions>
+        <execution>
+            <!-- ... -->
+            <configuration>
+                <!-- ... -->
+                <packaging-scripts>
+                    <packaging-script>
+                        <scripts>
+                            <script>infinispan.cli</script>
+                        </scripts>
+                    </packaging-script>
+                </packaging-scripts>
+            </configuration>
+        </execution>
+    </executions>
+</plugin>
+----
+
+== Ingress Configuration
+
+A https://kubernetes.io/docs/concepts/services-networking/ingress/[Kubernetes Ingress] is an object that manages external access to the services in a cluster, typically via HTTP.
+It provides load balancing, SSL termination and name-based virtual hosting.
+In order to support the affinity requirements described above, an Ingress Controller implementation, supporting L7 load balancing and exposes the requisite annotations for configuring session affinity, will need to be configured.
+The following sections will configure an https://haproxy-ingress.github.io/[HAProxy Ingress Controller] for optimal routing.
+
+=== Configuring Affinity in Deployment Image
+
+// TODO - update this section as we make this the default - https://issues.redhat.com/browse/CLOUD-4174
+
+First, we will need to reconfigure the application server to store the affinity information in a separate cookie.
+Previously WildFly only supported the ability to encode session affinity information into the JSESSIONID cookie, which is already used to uniquely identify a session.
```

Comment:
Rather than state the limitations of older versions, I think it would be clearer to indicate that the default behavior encodes the affinity into the existing JSESSIONID cookie.

Reply:
Fixed.